### PR TITLE
Add log operators

### PR DESCRIPTION
**Pull request by PaigeM89**
## Proposed Changes

Adds some operators to help create & add to logs in a condensed manner.

## Types of changes

What types of changes does your code introduce to FsLibLog?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] Build and tests pass locally
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added necessary documentation (if appropriate)

## Further comments

The operators are based on our discussions & usage. As it would be a pain to change them later, it might be worthwhile to bikeshed them now.

I am not sure how much description to add to the comments on the operators themselves. Ideally we would mirror the functions they're wrapping, but I worry the comments would be out-of-sync at some point.

### CODE REVIEW

File: src/FsLibLog/FsLibLog.fs [hunk 1]
```diff
@@ -789,3 +789,54 @@ module LogProvider =
         let stackFrame = StackFrame(2, false)
         getLoggerByType(stackFrame.GetMethod().DeclaringType)
 
+/// Provides operators to make writing logs more streamlined.
+module Operators =
+
+    /// Initializes a log with the specified message
```
Comment (baronfel) [inline src/FsLibLog/FsLibLog.fs:795]: please don't use the old markdown-style comments, it just causes pain for non-VS Code editors :(  Ionide should be able to generate properly-formatted xmldoc blocks for you these days though to help make the process smoother.

File: src/FsLibLog/FsLibLog.fs [hunk 2]
```diff
@@ -789,3 +789,54 @@ module LogProvider =
...
+    /// <param name="key">The name for the parameter.</param>
+    /// <param name="value">The value for the parameter.</param>
+    /// <returns>The amended log with the parameter added.</returns>
+    let (>>!-) log (key, value) = log >> Log.addContext key value
+
+    /// <summary>
+    /// Amends a Log with additional named parameters for context. This helper adds more context to a log. This DOES NOT affect the parameters set for a message template.
+    /// This is the same calling OpenMappedContext right before logging. This destructures an object rather than calling ToString() on it. WARNING: Destructring can be expensive.
+    ///
+    /// Wrapper for `Log.addContextDestructured`.
```
Comment (TheAngryByrd) [inline src/FsLibLog/FsLibLog.fs:826]: I think you'll want to use the `<see>` tag to get the link setup correctly https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments#see

Comment (PaigeM89) [reply src/FsLibLog/FsLibLog.fs]: Neither Ionide (mostly working) nor Rider (no XML handling in F#, apparently) provided a suitable testing ground to ensure these work 100%, but following [the docs](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/), I _think_ I got it right?

**Comment by TheAngryByrd** [comment]
Thanks!

**Pull request merged**